### PR TITLE
feat: automate loading arguments

### PR DESCRIPTION
**Pull request by johannaSommer**
## Description
This PR aims to eliminate a common error, where the base model is loaded in a specific precision but the user forgets to specify this precision again when loading the smashed model. This can also lead to wrong results for the `GPUMemory` metric. 
Generally I've updated the documentation to remove guidance on loading arguments, as the most important arguments are covered:
- torch_dtype is handled automatically now based on the base model
- device is handled via the smash config device (even though in the future, for multi-gpu support, this has to be improved)
- other arguments like e.g. deactivating the safety checked for image generation are automatically ported to the new model

## Related Issue
None.

## Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## How Has This Been Tested?
Tested by running:
```python
import torch
from transformers import AutoModelForCausalLM, AutoTokenizer
from pruna import PrunaModel

model_id = "meta-llama/Llama-3.2-1B"
base_model = AutoModelForCausalLM.from_pretrained(model_id, torch_dtype=torch.bfloat16)
base_model = PrunaModel(base_model)
base_model.save_pretrained("saved_llm")
loaded_model = PrunaModel.from_pretrained("saved_llm")
print(loaded_model.lm_head.weight.dtype)
```
```bash
>> torch.bfloat16
```

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Additional Notes
None.

### CODE REVIEW

**Review by begumcig — APPROVED** [review]
Amazing!!!!!

**Review by johnrachwan123 — APPROVED** [review]
LGTM!
